### ISSUE TITLE
JDK8 Optional not working with Factory builder

### ISSUE THREAD

**Issue by fzakaria** [immutables/immutables#262]
```
    @Builder.Factory
    public static String fakeMethod(
            Optional<String> input,
            Optional<Details> details,
            Optional<ChildPolicy> childPolicy) {
        return "Hello World";
    }
```

> [ERROR] /Users/fzakaria/Development/WaterFlow/target/generated-sources/annotations/com/github/fzakaria/waterflow/FakeMethodBuilder.java:[105,35] incompatible types: java.lang.String cannot be converted to java.util.Optional<java.lang.String>

```
@SuppressWarnings("all")
@ParametersAreNonnullByDefault
@Generated({"Immutables.generator", "SwfRequests.fakeMethod"})
@NotThreadSafe
public final class FakeMethodBuilder {
  private @Nullable String input;
  private @Nullable Details details;
  private @Nullable ChildPolicy childPolicy;
```

If I set the `Optionals` to that from `Guava` the problem does not exist

**Comment by elucash** [comment]
yes, found a problem, thank you!

**Comment by elucash** [comment]
Released in 2.1.11. Thanks!
